Assets/Scripts/ExplotionController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Mathematics;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.VFX;
+
+public class ExplotionController : MonoBehaviour
+{
+    [SerializeField] 
+    public int damage;
+
+    [SerializeField]
+    private VisualEffect visualEffect;
+
+    [SerializeField]
+    public float radius;
+
+    [SerializeField]
+    private LayerMask hitBoxLayers;
+    
+    private HashSet<HealthController> hitHealthControllers = new HashSet<HealthController>();
+    
+
+    private void Start()
+    {
+        visualEffect.enabled = false;
+    }
+
+    public void OnDrawGizmos()
+    {
+        Gizmos.DrawWireSphere(transform.position,radius);
+    }
+
+    public void Explode()
+    {
+        visualEffect.enabled = true;
+        visualEffect.SendEvent("OnPlay");
+        Collider[] colliderList = Physics.OverlapSphere(transform.position, radius, hitBoxLayers);
+        
+        foreach (Collider collider in colliderList)
+        {
+            HitOrMiss(collider);
+        }
+    }
+
+    private void HitOrMiss(Collider collider)
+    {
+        HitboxController controller = collider.GetComponent<HitboxController>();
+        if (!controller.health || !hitHealthControllers.Contains(controller.health))
+        {
+            hitHealthControllers.Add(controller.health);
+            float distdam = math.pow((Vector3.Distance(collider.transform.position, transform.position) / radius), 2);
```

Comment:
Please rephrase this variable – I get what you mean, but it could be written in a more clear, verbose way. Abbreviations should be avoided in this context. (and do note that damage is not relevant to this variable, it's more like a scale factor based on distance)